tests/negotiation.rs

```diff
@@ -129,47 +123,70 @@ fn serialize_public_offer() {
+               0000000000000000000000000000000000000000000000000000000260700";
+    let res: PublicOffer<bitcoin::Amount, monero::Amount, CSVTimelock, SatPerVByte> =
+        strict_encoding::strict_deserialize(&hex::decode(hex).unwrap()).unwrap();
+    // same fingerprint
+    assert_eq!(id, res.fingerprint());
+}
+
+#[test]
+fn get_public_offer_uuid() {
+    let hex = "46435357415001004450e567b1106f429247bb680e5fe0c80200000080800000800800a08601000\
+               00000000800c80000000000000004000a00000004000a0000000108001400000000000000022100\
+               03b31a0a70343bb46f3db3768296ac5027f9873921b37f852860c690063ff9e4c90000000000000\
+               0000000000000000000000000000000000000000000000000000000260700";
     let res: PublicOffer<bitcoin::Amount, monero::Amount, CSVTimelock, SatPerVByte> =
         strict_encoding::strict_deserialize(&hex::decode(hex).unwrap()).unwrap();
-    let id =
-        PublicOfferId::from_str("3a466a0a0cff7bf800808653460076549621d07db78e697b9dfaebaba0ab8b33")
-            .unwrap();
-    assert_eq!(id, res.id());
+    assert_eq!(uuid!("67e55044-10b1-426f-9247-bb680e5fe0c8"), res.uuid());
 }
 
 #[test]
 fn check_public_offer_magic_bytes() {
-    let valid = "46435357415001000200000080800000800800a0860100000000000800c80000000000000004000\
-                 a00000004000a000000010800140000000000000002210003b31a0a70343bb46f3db3768296ac50\
-                 27f9873921b37f852860c690063ff9e4c9000000000000000000000000000000000000000000000\
-                 00000000000000000000000260700";
+    let valid = "46435357415001004450e567b1106f429247bb680e5fe0c80200000080800000800800a08601000\
+                 00000000800c80000000000000004000a00000004000a0000000108001400000000000000022100\
+                 03b31a0a70343bb46f3db3768296ac5027f9873921b37f852860c690063ff9e4c90000000000000\
+                 0000000000000000000000000000000000000000000000000000000260700";
     let pub_offer: Result<
         PublicOffer<bitcoin::Amount, monero::Amount, CSVTimelock, SatPerVByte>,
         consensus::Error,
     > = deserialize(&hex::decode(valid).unwrap()[..]);
     assert!(pub_offer.is_ok());
 
-    let invalid = "474353574150010002000000808000008008a08601000000000008c800000000000000040a00000\
-                   0040a0000000108140000000000000002";
+    let inval = "47435357415001004450e567b1106f429247bb680e5fe0c80200000080800000800800a08601000\
+                 00000000800c80000000000000004000a00000004000a0000000108001400000000000000022100\
+                 03b31a0a70343bb46f3db3768296ac5027f9873921b37f852860c690063ff9e4c90000000000000\
+                 0000000000000000000000000000000000000000000000000000000260700";
     let pub_offer: Result<
         PublicOffer<bitcoin::Amount, monero::Amount, CSVTimelock, SatPerVByte>,
         consensus::Error,
-    > = deserialize(&hex::decode(invalid).unwrap()[..]);
+    > = deserialize(&hex::decode(inval).unwrap()[..]);
```

Comment:
invalid?

Reply:
Was to keep the same number of chars as 'valid' for the alignment :)

Done in dae445940260b6008766f1edc2e4de01bb769fdc